src/pages/RealmExplorer.js

```diff
@@ -3,201 +3,214 @@
-
-      // TODO: use explicit loops instead of generators.
-      // eslint-disable-next-line no-restricted-syntax
-      for (const [key] of Object.entries(firstObject)) {
-        headerCells.push(
-          <HeaderCell
-            key={key}
-            style={globalStyles.headerCell}
-            textStyle={globalStyles.text}
-            width={1}
-            text={key}
-          />
-        );
-      }
-    }
-    return <Header style={globalStyles.header}>{headerCells}</Header>;
-  }
+  const onSearchChange = searchString => {
+    setState({ ...state, searchString });
+  };
 
-  renderRow(item) {
-    const { data } = this.state;
-    const cells = [];
-    if (data && data.length > 0) {
-      const firstObject = data[0];
-
-      const objects = Object.entries(firstObject);
-
-      // TODO: use explicit loops instead of generators.
-      // eslint-disable-next-line no-restricted-syntax
-      for (const [key] of objects) {
-        let itemString =
-          item[key] &&
-          (typeof item[key] === 'string' ||
-            typeof item[key] === 'number' ||
-            typeof item[key].getMonth === 'function') &&
-          String(item[key]);
-        if (!itemString && item[key] && item[key].length) {
-          itemString = item[key].length;
-        }
-        if (typeof item[key] === 'boolean') {
-          itemString = item[key] ? 'True' : 'False';
-        }
-        if (!itemString && item[key] && item[key].id) itemString = item[key].id;
-        cells.push(
-          <Cell key={key} style={globalStyles.cell} textStyle={globalStyles.text} width={1}>
-            {itemString}
-          </Cell>
-        );
-      }
-    }
-    return <Row style={globalStyles.row}>{cells}</Row>;
-  }
+  const onFilterChange = filterString => {
+    setState({ ...state, filterString });
+  };
+
+  useEffect(() => setState(getUpdatedState(database, state)), [
+    state.searchString,
+    state.filterString,
+  ]);
 
-  render() {
-    const { dataSource } = this.state;
-
-    return (
-      <View style={[globalStyles.container]}>
-        <SearchBar onChange={this.onSearchChange} placeholder="Table Name" />
-        <SearchBar onChange={this.onFilterChange} placeholder="Filter" />
-        <DataTable
-          style={globalStyles.container}
-          listViewStyle={globalStyles.container}
-          dataSource={dataSource}
-          renderRow={this.renderRow}
-          renderHeader={this.renderHeader}
-        />
+  const objectFields = OBJECT_FIELDS[state.objectString];
+
+  const headerRowKeyExtractor = (_, index) => index.toString();
+  const rowKeyExtractor = ({ id }) => id;
+
+  const renderHeaderRow = headerRow => {
+    const { item } = headerRow;
+    const cells = Object.keys(item).map(columnKey => (
+      <View style={styles.cell}>
+        <Text style={styles.cellText}>{columnKey}</Text>
       </View>
-    );
-  }
-}
+    ));
+    return <View style={styles.row}>{cells}</View>;
+  };
+
+  const renderRow = row => {
```

Comment:
I probably wouldn't bother but this is a good candidate for `useCallback` will memoize and existing rows when you filter